pkg/utils/utils.go

```diff
@@ -1105,3 +1110,101 @@ func DetermineUnitTestRun(ctx context.Context) bool {
 	}
 	return unitTestRun
 }
+
+func BrownfieldDeployment(ctx context.Context, path string, cr csmv1.ApexConnectivityClient, ctrlClient crclient.Client) error {
+	//Get the namespaces
+	log := logger.GetLogger(ctx)
+
+	namespace, err := GetNamespaces(ctx)
+	if err != nil {
+		log.Error(err, "Failed to get namespaces")
+		return err
+	}
+
+	buf, err := os.ReadFile(filepath.Clean(path))
+	if err != nil {
+		log.Error(err, "Failed to read manifest file")
+		return err
+	}
+
+	yamlFile := string(buf)
+
+	for _, ns := range namespace {
+
+		yamlFile := strings.ReplaceAll(yamlFile, ExistingNamespace, ns)
+		yamlFile = strings.ReplaceAll(yamlFile, ClientNamespace, cr.Namespace)
+
+		err := CreateObjects(ctx, yamlFile, ctrlClient)
+		if err != nil {
+			return err
+		}
+	}
+	log.Infoln("Brownfield deploymet success")
+	return nil
+}
+
+// Get the namespaces of csm-objects in the cluster
+func GetNamespaces(ctx context.Context) ([]string, error) {
+	log := logger.GetLogger(ctx)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// Create a Kubernetes clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Define label selector
+	labelSelector := "csm"
+
+	// Get all pods in the cluster
+	pods, err := clientset.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		log.Error(err, "Failed to get pods in the cluster")
+
+		os.Exit(1)
+	}
+
+	// Set to store unique namespaces
+	namespaceSet := make(map[string]struct{})
+
+	// Using map so that duplicate namespace entries are not added
+	for _, pod := range pods.Items {
+		namespaceSet[pod.Namespace] = struct{}{}
+	}
+
+	// Convert set to slice
+	var namespaces []string
+	for namespace := range namespaceSet {
+		namespaces = append(namespaces, namespace)
+	}
+
+	return namespaces, nil
+}
+
+func CreateObjects(ctx context.Context, yamlFile string, ctrlClient crclient.Client) error {
+	deployObjects, err := GetModuleComponentObj([]byte(yamlFile))
+	if err != nil {
+		return err
+	}
+
+	for _, obj := range deployObjects {
+		log.FromContext(ctx).Info("namespace of parsed object is", "object", obj.GetNamespace())
+
+		if err := ctrlClient.Create(ctx, obj); err != nil {
+			if k8serror.IsAlreadyExists(err) {
+				log.FromContext(ctx).Info("Object already exists", "object", obj.GetObjectKind().GroupVersionKind().String())
+				continue
+			}
+			return fmt.Errorf("failed to create object: %w", err)
```

Comment:
What if an object with the name exists but is in a different state than what we need to create? Should an update be done? Also at least log at warning level.

Reply:
Addressed this